experimental/packages/otlp-exporter-base/src/platform/browser/OTLPExporterBrowserBase.ts

```diff
@@ -73,7 +78,7 @@ export abstract class OTLPExporterBrowserBase<
     const body = JSON.stringify(serviceRequest);
 
     const promise = new Promise<void>((resolve, reject) => {
-      if (this._useXHR) {
+      if (this.sendMethod === 'xhr') {
```

Comment:
So if we  now know that the body is >= 64kb  then we also know that `sendBeacon` or `fetch (with the keep-alive flag set)` will never send the payload -- so we should pre-select these out as options.

Reply:
Yes, you're correct.
However, It's from the original code, and I'm not focused at that point.

I agree that the sender should be refactored a bit more for safety and matching such restrictions.
I think sendBeacon may not be the first choice, and fallback is needed even if sendBeacon is the primary method.

For example, just checking body size  < 64kb in the line is NOT safe. Another more complex way is better for some reasons (e.g. should check the return value of `sendBeacon` and fallbacks to another method).
First, Chrome checks the total size of payloads by all `sendBeacon` calls.
Second, the limit value may be changed in the future, and there may be various limit values per user-agent.

On the other hand, I think the current way is worked in general cases.